lib/llhttp/parser.js

```diff
@@ -0,0 +1,240 @@
+'use strict'
+
+/* global WebAssembly */
+
+const { resolve } = require('path')
+const { readFileSync } = require('fs')
+const constants = require('./constants')
+const { kMaxHeadersSize } = require('../core/symbols')
+const WASM_BUILD = resolve(__dirname, './llhttp.wasm')
+const bin = readFileSync(WASM_BUILD)
+const mod = new WebAssembly.Module(bin)
+
+const kOnMessageBegin = 0
+const kOnHeaders = 1
+const kOnHeadersComplete = 2
+const kOnBody = 3
+const kOnMessageComplete = 4
+const kOnExecute = 5
+
+const kPtr = Symbol('kPrt')
+const kStatusMessage = Symbol('kStatusMessage')
+const kHeadersFields = Symbol('kHeadersFields')
+const kHeadersValues = Symbol('kHeadersValues')
+const kBufferSize = Symbol('kBufferSize')
+const kHeadersTimeout = Symbol('kHeadersTimeout')
+const kBufferPtr = Symbol('kBufferPtr')
+const kMallocBuffer = Symbol('kMallocBuffer')
+const kCurrentBuffer = Symbol('kCurrentBuffer')
+const kGetHeaders = Symbol('kGetHeaders')
+const kMakeError = Symbol('kMakeError')
+
+/**
+ * Map of HTTPParser instances
+ */
+const instMap = new Map()
```

Comment:
I'm somewhat concerned about this map and wasm resources leaking out.

We are also accessing it every single time there is a callback from wasm, which adds overhead.

Given that every single callback from wasm is synchronous, maybe we could store only the current parser as a module-global and clear it up after execute. This should speed things up and avoid potential leakings.

Reply:
Ok, I can try that approach.

Reply:
One thing though, are you suggesting getting rid of the map also? Or just store the parser at the beginning of the message and cleaning at the end?

Reply:
```js

execute (data) {
  currentParser = this
  ...
  currentParser = false
}

Reply:
You can probably remove the map entirely